### ISSUE TITLE
R API h2o4gpu::install_h2o4gpu missing

### ISSUE THREAD

**Issue by jtelleria** [h2oai/h2o4gpu#530]
One of the points it was discussed when it was decided to develop the h2o4gpu R API through RStudio's reticulate R package was to provide a function for installing the python environment easily from R, but such function in this first release of the h2o4gpu R Package is missing.

I would propose to develop such function: h2o4gpu::install_h2o4gpu():

https://cran.r-project.org/web/packages/keras/keras.pdf

https://github.com/rstudio/keras/blob/master/R/install.R

**Comment by terrytangyuan** [comment]
Currently the s3 URLs for the releases do not look like what I expected, e.g. some questions I had in Slack channel:
- Why is there `0.2-nonccl` when this is for `0.2.0.9999` - `"https://s3.amazonaws.com/h2o-release/h2o4gpu/releases/bleeding-edge/ai/h2o/h2o4gpu/0.2-nonccl-cuda8/h2o4gpu-0.2.0.9999-cp36-cp36m-linux_x86_64.whl"` (edited)

- Should `0.2-nccl-cuda9` in `https://s3.amazonaws.com/h2o-release/h2o4gpu/releases/stable/ai/h2o/h2o4gpu/0.2-nccl-cuda9/h2o4gpu-0.2.0-cp36-cp36m-linux_x86_64.whl` be `0.2.0-nccl-cuda9` instead?

I propose not to implement a `install_h2o4gpu()` function, instead [these methods](https://rstudio.github.io/reticulate/reference/index.html#section-package-installation) from reticulate should suffice as well as the most recently added [py_install()](https://rstudio.github.io/reticulate/reference/py_install.html) method. You can simply provide the S3 url and the wheels can be easily installed.

**Comment by jtelleria** [comment]
h2o4gpu shall follow, in my opinion, the "K.I.S.S." Principle ("Keep it simple, stupid") in order to reach a wide audience and be a success within the R Community. 

Therefore, I would:
* Include these reticulate convenience functions (py_install, use_virtualenv, etc) into the h2o4gpu package NAMESPACE, so that R users only have to take care of h2o4gpu, and not reticulate.
* Include a function in h2o4gpu that checks that the h2o4gpu Version is the same than the h2o4gpu R Version, which shall be launched every time you start using h2o4gpu (Just as "h2o" R Package Checks Java).

Indeed @terrytangyuan, If you feel that we are not sure which version include (https://s3.amazonaws.com/h2o-release/h2o4gpu/releases/stable/ai/h2o/h2o4gpu/0.2-nccl-cuda9/h2o4gpu-0.2.0-cp36-cp36m-linux_x86_64.whl?) would not a regular R users who just want to test the library, feel overwhelmed with his first impression of the package (for having a complicated installation) and decide to give up?

h2o4gpu::install_h2o4gpu() shall only be a wrapper of the new py_install() with the correct version of the Python Package for each distro.

**Comment by terrytangyuan** [comment]
@jtelleria Yes, we currently reexported those methods, except for `py_install` since we just added that in reticulate very recently. 

Good points indeed. I think we'll have to wait until those S3 URLs are fixed or until we publish our wheels to PyPI (cc: @mdymczyk). Thanks!
